contracts/pcv/curve/StableSwapOperatorV1.sol

```diff
@@ -0,0 +1,264 @@
+    function get_virtual_price() external view returns (uint256);
+    function calc_withdraw_one_coin(uint256 _token_amount, int128 i) external view returns (uint256);
+}
+interface I3Pool {
+    function add_liquidity(uint256[3] memory amounts, uint256 min_mint_amount) external;
+}
+
+/// @title StableSwapOperator: implementation for a Curve Metapool market-maker.
+/// This is version 1, it only allows simple deposits and does not require the
+/// Minter role. FEI has to be minted on this contract before calling deposit().
+/// There are no reweight mechanisms.
+/// All accounting and withdrawals are denominated in DAI. Should the contract
+/// hold USDC or USDT, these can also be deposited.
+/// @author eswak
+contract StableSwapOperatorV1 is PCVDeposit {
+    using SafeERC20 for ERC20;
+
+    // ------------------ Properties -------------------------------------------
+
+    uint256 public depositMaxSlippageBasisPoints;
+    uint256 public constant BASIS_POINTS_GRANULARITY = 10_000;
+
+    /// @notice The StableSwap pool to deposit in
+    address public pool;
+
+    // ------------------ Private properties -----------------------------------
+
+    /// some fixed variables to interact with the pool
+    uint8 private _feiIndex; // index of FEI in the pool (0 or 1)
+    uint8 private _3crvIndex; // index of 3crv in the pool (0 or 1)
+    address private _3crv; // address of the 3crv token
+    address private _3pool; // address of the 3pool
+    address private _dai; // address of the DAI token
+    address private _usdc; // address of the USDC token
+    address private _usdt; // address of the USDT token
+
+    // ------------------ Constructor ------------------------------------------
+
+    /// @notice Curve PCV Deposit constructor
+    /// @param _core Fei Core for reference
+    /// @param _pool StableSwap to deposit to
+    /// @param _curve3pool the Curve 3pool
+    /// @param _depositMaxSlippageBasisPoints max slippage for deposits, in bp
+    constructor(
+        address _core,
+        address _pool,
+        address _curve3pool,
+        uint256 _depositMaxSlippageBasisPoints
+    ) CoreRef(_core) {
+        // public variables
+        pool = _pool;
+        depositMaxSlippageBasisPoints = _depositMaxSlippageBasisPoints;
+
+        // cached private variables
+        _3crvIndex = IStableSwap(pool).coins(0) == address(fei()) ? 1 : 0;
+        _feiIndex = _3crvIndex == 0 ? 1 : 0;
+        _3crv = IStableSwap(pool).coins(_3crvIndex);
+        _3pool = _curve3pool;
+        _dai = IStableSwap(_3pool).coins(0);
+        _usdc = IStableSwap(_3pool).coins(1);
+        _usdt = IStableSwap(_3pool).coins(2);
+    }
+
+    /// @notice deposit DAI, USDC, USDT, 3crv, and FEI into the pool.
+    /// Note: the FEI has to be minted & deposited on this contract in a previous
+    ///       tx, as this contract does not use the Minter role.
+    function deposit() public override onlyPCVController whenNotPaused {
```

Comment:
why onlyPCVController? Normally we allow any address to deposit

Reply:
it used to be able to mint FEI so I wanted stricter access control, but I guess we can remove it now